crates/subspace-farmer/src/bin/subspace-farmer/main.rs

```diff
@@ -26,10 +26,9 @@ struct FarmingArgs {
     #[clap(long)]
     bootstrap_nodes: Vec<Multiaddr>,
     /// Multiaddr to listen on for subspace networking, for instance `/ip4/0.0.0.0/tcp/0`,
-    /// multiple are supported, subspace networking will pick ports automatically if nothing is
-    /// specified.
+    /// subspace networking will pick ports automatically if nothing is specified.
     #[clap(long)]
-    dsn_ports_start: Option<u16>,
+    listen_on: Option<Multiaddr>,
```

Comment:
Why is it still an `Option`? It should be `Vec` just like before so you can specify what interfaces you want to listen on, with one parameter you can select one interface or all interfaces, but not multiple.

Reply:
And for each parameter we add an increment to a port? Am I getting it right?

Reply:
BTW I think it is worth setting default here on CLI level, so we don't need to unwrap it in the code below and we don't need to explain it explicitly here.

Reply:
> And for each parameter we add an increment to a port? Am I getting it right?

Exactly.